tests/test_coupons.py

```diff
@@ -122,6 +123,20 @@ def test_delete(self):
             coupon.refresh_from_db()
             self.assertTrue(coupon.is_deleted)
 
+        with mock.patch("aa_stripe.models.StripeCoupon.delete") as mocked_delete:
+            # also test the overriden queryset's delete
+            coupon2 = self._create_coupon(coupon_id="CPON2")
+            coupon3 = self._create_coupon(coupon_id="CPON3")
+            coupons_qs = StripeCoupon.objects.filter(pk__in=[coupon2.pk, coupon3.pk])
+            self.assertEqual(StripeCoupon.objects.filter(is_deleted=False).count(), 2)
+            self.assertEqual(coupons_qs.count(), 2)
+            delete_result = coupons_qs.delete()
+            self.assertEqual(mocked_delete.call_count, 2)
+            self.assertEqual(delete_result, (0, {"aa_stripe.StripeCoupon": 0}))
+            self.assertEqual(coupons_qs.count(), 2)
+            self.assertEqual(StripeCoupon.objects.filter(is_deleted=True).count(), 0)
```

Comment:
self.assertEqual(delete_result, (0, {"aa_stripe.StripeCoupon": 0}))
            self.assertEqual(coupons_qs.count(), 2)
>           self.assertEqual(StripeCoupon.objects.filter(is_deleted=True).count(), 0)
E           AssertionError: 1 != 0
Why? I tested also between lines 130 and 131 by adding self.assertEqual(StripeCoupon.objects.filter(is_deleted=True).count(), 1) and it failed, so since you deleted two coupons there should be two with is_deleted=True not 1.